go-chaos/internal/dataloss_sim_helper.go

```diff
@@ -0,0 +1,174 @@
+*/
+func (c K8Client) ApplyInitContainerPatch() error {
+	// apply config map
+	err := createConfigMapForInitContainer(c)
+	if err != nil {
+		fmt.Printf("Failed to create config map %s", err)
+		return err
+	}
+
+	statefulSet, err := c.GetZeebeStatefulSet()
+	if err != nil {
+		fmt.Printf("Failed to get statefulset %s", err)
+		return err
+	}
+
+	// Adds init container patch
+	patch := []byte(`{
+  "spec": {
+    "template": {
+      "spec": {
+        "volumes": [
+          {
+            "name": "zeebe-control-pod-restart-flags-mount",
+            "configMap": {
+              "name": "zeebe-control-pod-restart-flags"
+            }
+          }
+        ],
+        "initContainers": [
+          {
+            "name": "busybox",
+            "image": "busybox:1.28",
+            "command": [
+              "/bin/sh",
+              "-c"
+            ],
+            "args": [
+              "while true; do block=$(cat /etc/config/block_${K8S_NAME##*-}); if [ $block == \"false\" ]; then break; fi; sleep 10; done"
+            ],
+            "env": [
+              {
+                "name": "K8S_NAME",
+                "valueFrom": {
+                  "fieldRef": {
+                    "fieldPath": "metadata.name"
+                  }
+                }
+              }
+            ],
+            "volumeMounts": [
+              {
+                "name": "zeebe-control-pod-restart-flags-mount",
+                "mountPath": "/etc/config"
+              }
+            ]
+          }
+        ]
+      }
+    }
+  }
+}`)
+	_, err = c.Clientset.AppsV1().StatefulSets(c.GetCurrentNamespace()).Patch(context.TODO(), statefulSet.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
+	if err != nil {
+		fmt.Printf("Failed to apply init container patch %s", err)
+		return err
+	}
+	if Verbosity {
+		fmt.Printf("Applied init container patch to %s \n", statefulSet.Name)
+	}
+	return err
+}
+
+func createConfigMapForInitContainer(c K8Client) error {
+	cm, err := c.Clientset.CoreV1().ConfigMaps(c.GetCurrentNamespace()).Get(context.TODO(), configMapName, metav1.GetOptions{})
+	if err == nil {
+		fmt.Printf("Config map %s already exists. Will not create again. \n", cm.Name)
+		return nil
+	}
+
+	if k8sErrors.IsNotFound(err) {
+		// create config map
+		cm := corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ConfigMap",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      configMapName,
+				Namespace: c.GetCurrentNamespace(),
+			},
+
+			Data: map[string]string{
+				// When set to true the corresponding zeebe pods will be prevented from starting up.
+				// It will be blocked in the Init container until this flag is set back to false.
+				"block_0": "false",
+				"block_1": "false",
+				"block_2": "false",
+				"block_3": "false",
+				"block_4": "false",
+				"block_5": "false",
+				"block_6": "false",
+				"block_7": "false",
+				"block_8": "false",
+			},
```

Comment:
:thought_balloon: :wrench: Might make sense to parameterize this or do it based on the topology? :thinking:

Reply:
I had thought about it. But I don't want to do it now. Now our tests are at max with 6 nodes, so here upto 8 nodeId would be enough. When we need to test on larger cluster we can then modify it to make it more flexible based on the topology.